sceptre/plan/actions.py

```diff
@@ -614,18 +614,130 @@ def diff(self, differ="difflib"):
         local_template_stream = self.stack.template.body
 
         if differ == "difflib":
-            return_val = self._diff_by_difflib(
+            response = self._diff_by_difflib(
                 remote_template_stream,
                 local_template_stream
             )
 
         elif differ == "dictdiffer":
-            return_val = self._diff_by_dictdiffer(
+            response = self._diff_by_dictdiffer(
                 remote_template_stream,
                 local_template_stream
             )
 
-        return [self.stack.external_name, return_val]
+        return [self.stack.external_name, response]
+
+    @add_stack_hooks
```

Comment:
Can you think of a usecase where we need hooks?

Reply:
Nope, removed.